Keen/EventCollection.cs

```diff
@@ -1,100 +1,158 @@
+            if (!responseMsg.IsSuccessStatusCode)
+            {
+                throw new KeenException("GetSchema failed with status: " + responseMsg.StatusCode);
             }
+
+            return response;
         }
 
-        public async System.Threading.Tasks.Task AddEvent(string collection, JObject anEvent)
+        public async Task DeleteCollection(string collection)
         {
-            var content = anEvent.ToString();
+            if (string.IsNullOrWhiteSpace(_masterKey))
+            {
+                throw new KeenException("An API MasterKey is required to delete a collection.");
+            }
+
+            var responseMsg = await _keenHttpClient
+                .DeleteAsync(GetCollectionUrl(collection), _masterKey)
+                .ConfigureAwait(continueOnCapturedContext: false);
 
-            using (var client = new HttpClient())
-            using (var contentStream = new StreamContent(new MemoryStream(Encoding.UTF8.GetBytes(content))))
+            if (!responseMsg.IsSuccessStatusCode)
             {
-                contentStream.Headers.Add("content-type", "application/json");
-
-                client.DefaultRequestHeaders.Add("Authorization", _prjSettings.WriteKey);
-                client.DefaultRequestHeaders.Add("Keen-Sdk", KeenUtil.GetSdkVersion());
-
-                var httpResponse = await client.PostAsync(_serverUrl + collection, contentStream)
-                    .ConfigureAwait(continueOnCapturedContext: false);
-                var responseString = await httpResponse.Content.ReadAsStringAsync()
-                    .ConfigureAwait(continueOnCapturedContext: false);
-                JObject jsonResponse = null;
-                try
-                {
-                    // Normally the response content should be parsable JSON,
-                    // but if the server returned a 404 error page or something
-                    // like that, this will throw. 
-                    jsonResponse = JObject.Parse(responseString);
-                }
-                catch (Exception) 
-                { }
-
-                // error checking, throw an exception with information from the 
-                // json response if available, then check the HTTP response.
-                KeenUtil.CheckApiErrorCode(jsonResponse);
-                if (!httpResponse.IsSuccessStatusCode)
-                    throw new KeenException("AddEvent failed with status: " + httpResponse);
+                throw new KeenException("DeleteCollection failed with status: " + responseMsg.StatusCode);
             }
         }
 
-        public EventCollection(IProjectSettings prjSettings)
+        public async Task AddEvent(string collection, JObject anEvent)
         {
-            _prjSettings = prjSettings;
+            if (string.IsNullOrWhiteSpace(_writeKey))
+            {
+                throw new KeenException("An API WriteKey is required to add events.");
+            }
+
+            var content = anEvent.ToString();
```

Comment:
I probably copy/pasted that. `event` is a keyword in C#, so one has to name it `@event` as per the language [spec](https://msdn.microsoft.com/en-us/library/aa664670.aspx), which I've done elsewhere, so I'm ambivalent. Some people prefer to avoid `@...` variables, so I imagine that's why it was called `an...` in whatever code I copied it from.